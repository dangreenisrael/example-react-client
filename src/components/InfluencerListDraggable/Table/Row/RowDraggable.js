import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import RowCheckboxContainer from './RowCheckboxContainer';
import Tag from '../../../Tag'

const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))
export default class DraggableRow extends Component {
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {activeStyle, connectDragSource, activeColumn, data, headings, id, selected, removeTag} = this.props;
    const style = selected ? activeStyle : {};
    return connectDragSource(
        <tr style={style}>
          <td>
            <RowCheckboxContainer rowId={id}/>
          </td>
          {headings.map((heading, i)=>{
            const propName = heading.propertyName;
            const style = (propName === activeColumn) ? activeStyle: {};
            return <td style={style} key={`data-${i}`}>
              {data[propName]}
            </td>
          })}
          <td>
            {data.tags.map((tag, i)=>{
              return <Tag key={`tags-${i}`} title={tag} remove={removeTag}/>
            })}
          </td>
        </tr>
    );
  }
}
