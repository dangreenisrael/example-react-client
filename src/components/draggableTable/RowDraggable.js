import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import CheckboxRowContainer from './CheckboxRowContainer';
const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))

export default class DraggableRow extends Component {
  constructor(props){
    super(props);
    this.toggleChecked = this.toggleChecked.bind(this);
    this.state = {checked: false};
  }
  toggleChecked(status){
    this.setState({checked: status});
  }
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {connectDragSource, data, headings, id} = this.props;
    const style = this.state.checked ? {backgroundColor: "rgba(23, 42, 251, 0.34)" } : {};
    const cellData = headings.map(key =>{
      return data[key]
    });
    return connectDragSource(
        <tr style={style}>
          <td>
            <CheckboxRowContainer rowId={id} toggleChecked={this.toggleChecked}/>
          </td>
          {cellData.map((cell, i)=>{
            return <td key={i}>
              {cell}
            </td>
          })}
        </tr>
    );
  }
}
