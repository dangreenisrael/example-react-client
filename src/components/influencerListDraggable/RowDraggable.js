import React, {Component} from 'react';
import {DragSource} from 'react-dnd';
import RowCheckboxContainer from './RowCheckboxContainer';
import Tag from '../Tag'

const boxSource = {
  beginDrag(props) {
    return { id: props.id };
  },
};

@DragSource(props => "application", boxSource, (connect, monitor) => ({
  connectDragSource: connect.dragSource(),
  connectDragPreview: connect.dragPreview(),
  isDragging: monitor.isDragging(),
}))
export default class DraggableRow extends Component {
  render() {
    this.props.connectDragPreview(this.props.dragPreview);
    const {connectDragSource, data, headings, id, selected, removeTag} = this.props;
    const style = selected ? {backgroundColor: "rgba(23, 42, 251, 0.34)" } : {};
    const cellData = headings.map(key =>{
      return data[key]
    });
    return connectDragSource(
        <tr style={style}>
          <td>
            <RowCheckboxContainer rowId={id}/>
          </td>
          {cellData.map((cell, i)=>{
            return <td key={`data-${i}`}>
              {cell}
            </td>
          })}
          <td>
            {data.tags.map((tag, i)=>{
              return <Tag key={`tags-${i}`} title={tag} remove={removeTag}/>
            })}
          </td>
        </tr>
    );
  }
}
