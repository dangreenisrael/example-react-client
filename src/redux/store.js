
import { createStore, compose, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import logLastAction from './middleware/log-last-action';
import combinedReducer from './reducers/index';

const store = createStore(combinedReducer, compose(
    applyMiddleware(thunk, logLastAction),

    typeof window === 'object' && typeof window.devToolsExtension !== 'undefined'
        ? window.devToolsExtension() : (f) => f
));

export default store;