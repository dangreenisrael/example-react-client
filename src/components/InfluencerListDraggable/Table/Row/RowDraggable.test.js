import React from 'react';
import renderer from 'react-test-renderer';
import RowDraggable from './RowDraggable';
import {Provider} from 'react-redux';
import store from '../../../../redux/store';

const headings = ["firstName", "lastName", "userName"];
const influencer = {
  firstName: 'Dan',
  lastName: 'Green',
  userName: 'DGreen',
  tags: []
};

const OriginalRow = RowDraggable.DecoratedComponent;
const identity = el => el;

test('Link renders correctly', () => {
  const tree = renderer.create(
      <Provider store={store}>
        <OriginalRow
            name='test'
            connectDragSource={identity}
            connectDragPreview={identity}
            headings={headings}
            data={influencer}
        />
      </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
