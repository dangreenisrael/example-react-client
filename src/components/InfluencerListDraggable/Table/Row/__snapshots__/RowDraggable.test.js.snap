exports[`test Link renders correctly 1`] = `
<tr
  style={Object {}}>
  <td>
    <input
      checked={false}
      name={undefined}
      onChange={[Function]}
      type="checkbox" />
  </td>
  <td
    style={undefined} />
  <td
    style={undefined} />
  <td
    style={undefined} />
  <td />
</tr>
`;
